packages/e2e-test-utils/src/site-editor.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * WordPress dependencies
+ */
+import { visitAdminPage } from '@wordpress/e2e-test-utils';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * @typedef {import('puppeteer-core').ElementHandle} ElementHandle
+ */
+
+const SELECTORS = {
+	navigationPanel: {
+		backToDashboard:
+			'.edit-site-navigation-panel .edit-site-navigation-panel__back-to-dashboard',
+		goBack: '.components-navigation__back-button',
+		isOpenState: '.edit-site-navigation-toggle.is-open',
+		menuItem: ( label ) =>
+			`//div[contains(@class, "edit-site-navigation-panel")]//button[.//*[text()="${ label }"]]`,
+		open: '.edit-site-navigation-toggle__button',
+		panelContainer: '.edit-site-navigation-panel',
+	},
+	visualEditor: '.edit-site-visual-editor iframe',
+};
+
+/**
+ * Searches for an item in the navigation panel with the label provided and clicks it.
+ *
+ * @param {string} label The label to search the menu item for.
+ */
+export async function clickSiteEditorMenuItem( label ) {
+	const item = await getSiteEditorMenuItem( label );
+
+	if ( item ) {
+		await item.click();
+	} else {
+		throw new Error(
+			`Navigation item with label ${ label } was not found.`
+		);
+	}
+}
+
+/**
+ * Closes the site editor navigation panel if open
+ */
+export async function closeSiteEditorNavigationPanel() {
+	const { navigationPanel } = SELECTORS;
+
+	const isOpen = !! ( await page.$( navigationPanel.isOpenState ) );
+
+	if ( isOpen ) {
+		await page.click( navigationPanel.open );
+		await page.waitForSelector( navigationPanel.panelContainer, {
+			hidden: true,
+		} );
+	}
+}
+
+/**
+ * Skips the welcome guide popping up to first time users of the site editor
+ */
+export async function disableWelcomeGuide() {
```

Comment:
We probably don't want to export this function anymore and make it internal.

Reply:
Mm… do you think so? I thought perhaps it might still be useful for some people to have it separately. Maybe they need to disable the welcome guide after doing some other steps?

I'm actually not sure. I'm confident you know best all the possible workflows, so if you say it would be useless, I'll remove the export and the reference in the docs.

Reply:
In that case, let's add `site` in the function name. Since it only disables Site Editor welcome guides.

P.S. I don't think failing tests is related to this PR. See #38486 for details.